src/TimeModeling/Types/GeometryStructure.jl

```diff
@@ -265,16 +303,37 @@ function Geometry(data::Array{SegyIO.SeisCon,1}; key="source", segy_depth_key=""
 
     nsrc = length(data)
     container = Array{SegyIO.SeisCon}(undef, nsrc)
-    dt = Array{Float32}(undef, nsrc); nt = Array{Integer}(undef, nsrc); t = Array{Float32}(undef, nsrc)
     nrec = Array{Integer}(undef, nsrc)
     for j=1:nsrc
         container[j] = data[j]
-        dt[j] = data[j].blocks[1].summary["dt"][1]/1f3
-        nt[j] = data[j].ns
-        t[j] = (nt[j]-1)*dt[j]
         nrec[j] = key=="source" ? 1 : Int((data[j].blocks[1].endbyte - data[j].blocks[1].startbyte)/(240 + data[j].ns*4))
     end
-    return  GeometryOOC{Float32}(container,dt,nt,t,nrec,key,segy_depth_key)
+
+    # Prepare dtCell
```

Comment:
These can probably be wrapped in some utility function to avoid the duplicate code.

Reply:
done